### DISCUSSION TITLE
Parametrized excel formulas in conditional formatting rule

### DISCUSSION THREAD

**Discussion by anastasiya-belova** [jxlsteam/jxls#157] [Q&A] [answered]
Hello!

I want to use "Parametrized excel formulas" (as in example http://jxls.sourceforge.net/samples/param_formulas.html) inside the conditional formatting rule (see screenshots 1 and 2).

<img width="1277" alt="Screenshot 2021-12-22 at 15 24 23" src="https://user-images.githubusercontent.com/56701074/147093058-681f4d32-1be5-4d97-bd81-5946a7886f1a.png">
<img width="510" alt="Screenshot 2021-12-22 at 15 25 22" src="https://user-images.githubusercontent.com/56701074/147093066-d3367698-66b2-4136-b360-39b9327898b6.png">

But after saving rule my formula changes (see screenshot 3). And, of course, after processing this template by jxls, variable "bonus" not substituted in formula.
<img width="505" alt="Screenshot 2021-12-22 at 15 26 38" src="https://user-images.githubusercontent.com/56701074/147093070-0433a824-6a0b-4b0c-8210-e9940773853f.png">
 
Is it possible to use variables from java code inside formula on conditional formatting?

**Comment by SoltauFintel** [comment, marked as answer]
hi

I don't know if JXLS supports JEXL formulas at that place. I guess not.

- Is ${bonus} a fix value for the whole Excel file?
- Why do you write [...] ?? $[ ... ] instead??
- Workaround: write ${bonus} e.g. into cell AA1 and use AA1 in the formula.

You could debug PoiSheetData.java:86 (updateConditionalFormatting). If the formula appears there we could use the JEXL evaluator at that place. For adding a new feature we'll need a PR. See https://github.com/jxlsteam/jxls/blob/master/CONTRIBUTING.md

**Reply by anastasiya-belova** [reply]
Thank you for answering my question.
Yes, ${bonus} is a fixed value for this Excel file. And yes, I made a typo - [...] instead $[] 😕 
Suggested workaround works correctly, it's great!
I'll try to add new future not now, but when I have free time.
